src/utils/url/toCanonicalURL.ts

```diff
@@ -0,0 +1,19 @@
+import { parse } from 'url';
+import { trimSlashes } from '.';
+
+export const toCanonicalURL = (url: string) => {
+  const ulrObj = parse(decodeURIComponent(url));
+  if (!ulrObj.search) return ulrObj.pathname;
+  const path = trimSlashes(
+    (ulrObj.pathname || '').split('api/')?.[1] || ulrObj.pathname,
+  );
+  return `${path}?${ulrObj.search
```

Comment:
Changes:
- Fix typo (`ulrObj` -> `urlObj`)
- Use `new URL()` instead of deprecated `url.parse()`
- Remove redundant condition `(urlObj.pathname || '')` since `urlObj.pathname` is always a string.

```suggestion
  const urlObj = new URL(decodeURIComponent(url));
  if (!urlObj.search) return urlObj.pathname;
  const path = trimSlashes(
    urlObj.pathname.split('api/')?.[1] || urlObj.pathname,
  );
  return `${path}?${urlObj.search
```